### PR TITLE
PlayerJS: Refactor using flow diagram

### PR DESCRIPTION
**Pull request by rubenpingol-xibo**
relates to xibosignageltd/xibo-private#505

### CODE REVIEW

**Comment by rubenpingol-xibo** [comment]
TEST: /Layout END

**Comment by rubenpingol-xibo** [comment]
TEST: /Layout END

**Comment by rubenpingol-xibo** [comment]
TEST: /Layout END

**Comment by rubenpingol-xibo** [comment]
TEST: /Layout END

**Comment by rubenpingol-xibo** [comment]
TEST: /Layout END

File: modules/rss-ticker.xml [hunk 1]
```diff
@@ -226,7 +226,7 @@ if ((properties.takeItemsFrom === 'end' && properties.reverseOrder === 0) ||
         items = $(newItems);
     }
 }
-return items;
+return {dataItems: items};
```
Comment (maurofmferrao) [inline modules/rss-ticker.xml:229]: `calendar.xml` also has a `onDataLoad` that returns items, so before merging this PR to develop, you might want to merge develop to you local and add that

**Comment by rubenpingol-xibo** [comment]
TEST: /Layout END

**Review by maurofmferrao — DISMISSED** [review]
Looks good to me. Let's run some tests now to see what needs to be fixed, and we can review the final PR and merge it with develop. :)

**Comment by rubenpingol-xibo** [comment]
TEST: /Layout END

**Comment by rubenpingol-xibo** [comment]
The previous commit for calendar module is incorrect that's why I have this latest commit.

**Comment by rubenpingol-xibo** [comment]
TEST: /Layout END

**Review by maurofmferrao — APPROVED** [review]

**Pull request merged**